sisl/geometry.py

```diff
@@ -1019,27 +1021,451 @@ def overlap(self, other, eps=0.1, offset=(0., 0., 0.), offset_other=(0., 0., 0.)
+        >>> geom = geom.tile(5, 0).repeat(7, 1)
+
+        Sort according to :math:`x` coordinate
+
+        >>> geom.sort(axis=0)
+
+        Sort according to :math:`z`, then :math:`x` for each group created from first sort
+
+        >>> geom.sort(axis=(2, 0))
+
+        Sort according to :math:`z`, then first lattice vector
+
+        >>> geom.sort(axis=2, lattice=0)
+
+        Sort according to :math:`z` (ascending), then first lattice vector (descending)
+
+        >>> geom.sort(axis=2, ascend=False, lattice=0)
+
+        Sort according to :math:`z` (descending), then first lattice vector (ascending)
+        Note how integer suffixes has no importance.
+
+        >>> geom.sort(ascend0=False, axis=2, ascend1=True, lattice=0)
+
+        Sort only atoms ``range(1, 5)`` first by :math:`z`, then by first lattice vector
+
+        >>> geom.sort(axis=2, lattice=0, atom=np.arange(1, 5))
+
+        Sort two groups of atoms ``[range(1, 5), range(5, 10)]`` (individually) by :math:`z`
+
+        >>> geom.sort(axis=2, atom=[np.arange(1, 5), np.arange(5, 10)])
+
+        The returned sorting indices may be used for manual sorting. Note
+        however, that this requires one to perform a sorting for all atoms.
+        In such a case the following sortings are equal.
+
+        >>> geom0, atom0 = geom.sort(axis=2, lattice=0, ret_atom=True)
+        >>> _, atom1 = geom.sort(axis=2, ret_atom=True)
+        >>> geom1, atom1 = geom.sort(lattice=0, atom=atom1, ret_atom=True)
+        >>> geom2 = geom.sub(np.concatenate(atom0))
+        >>> geom3 = geom.sub(np.concatenate(atom1))
+        >>> assert geom0 == geom1
+        >>> assert geom0 == geom2
+        >>> assert geom0 == geom3
+
+        Default sorting is equivalent to ``axis=(0, 1, 2)``
+
+        >>> assert geom.sort() == geom.sort(axis=(0, 1, 2))
+
+        Sort along a user defined vector ``[2, 1, 0]``
+
+        >>> geom.sort(vector=[2, 1, 0])
+
+        Integer specification has no influence on the order of operations.
+        It is _always_ the keyword argument order that determines the operation.
+
+        >>> assert geom.sort(axis2=1, axis0=0, axis1=2) == geom.sort(axis=(1, 0, 2))
+
+        A too high `atol` may have unexpected side-effects. This is because of the way
+        the sorting algorithm splits the sections for nested sorting.
+        So for coordinates with a continuous displacement the sorting may break and group
+        a large range into 1 group. Consider the following array to be split in groups
+        while sorting.
+
+        >>> a = np.arange(5) * 0.01
+        >>> a[3:] += 0.1
+
+        For a high tolerance we have:
+
+        >>> (np.diff(a) > 0.02).nonzero()[0]
+        [2]
+
+        This would split the sorting into sections ``a[:3]`` and ``a[3:]``
+
+        >>> (np.diff(a) > 0.002).nonzero()[0]
+        [0, 1, 2, 3]
+
+        This would split the sorting into sections ``a[:1]``, ``a[1:2]``, ``a[2:3]``, ``a[3:4]`` and ``a[4:5]``
+
```

Comment:
Clearer now, thanks. Could you add a more "realistic" example with relevance of changing (raising) `atol`?

Minor stuff: I would suggest to move this `atol` discussion to after the `sort(group=...)`.

Reply:
Moved it to be the last entry. Hmm... I did this small thing:

```python
An example would be a linear chain with a middle point with a much shorter distance.

        >>> x = np.arange(5) * 0.1
        >>> x[3:] -= 0.095
        y = z = np.zeros(5)
        geom = si.Geometry(np.stack((x, y, z), axis=1))
        >>> geom.xyz[:, 0]
        [0.    0.1   0.2   0.205 0.305]

        In this case a high tolerance (``atol>0.005`) would group atoms 2 and 3
        together

        >>> geom.sort(atol=0.01, axis=0, ret_atom=True)[1]
        [[0], [1], [2, 3], [4]]

        However, a very low tolerance will not find these two as atoms close
        to each other.

        >>> geom.sort(atol=0.001, axis=0, ret_atom=True)[1]
        [[0], [1], [2], [3], [4]]
```